model_analyzer/monitor/dcgm/dcgm_monitor.py

```diff
@@ -71,14 +71,15 @@ def __init__(self, gpus, frequency, tags, dcgmPath=None):
 
         frequency = int(self._frequency * 1000)
         fields = []
-        for tag in tags:
-            if tag in self.model_analyzer_to_dcgm_field:
-                dcgm_field = self.model_analyzer_to_dcgm_field[tag]
+        for metric in metrics:
+            if metric in self.model_analyzer_to_dcgm_field:
+                dcgm_field = self.model_analyzer_to_dcgm_field[metric]
```

Comment:
What is going on here? This has a code smell to it. It is almost like this should be a visitor, or something similar.

Reply:
I think its checking to make sure that the metric is registered with the DCGM monitor. I will remove the if, and change it to a try/except